python/ray/data/datasource/file_datasink.py

```diff
@@ -94,7 +95,17 @@ def on_write_start(self) -> None:
         """
         from pyarrow.fs import FileType
 
-        if self.try_create_dir:
+        # We should skip creating directories in s3 unless the user specifically
+        # overrides this behavior. PyArrow's s3fs implementation might collide
+        # with restrictive IAM permissions and skipping the create_dir call will
```

Comment:
> "PyArrow's s3fs implementation might collide with restrictive IAM permissions"

Could you elaborate on this in the source code? Don't think I would've understood this without our offline conversation

Reply:
Yeah, will update to make this more specific.